blocks/extensions/__init__.py

```diff
@@ -524,10 +524,14 @@ def __setstate__(self, state):
     def get_iter_per_epoch(self):
         """Try to infer the number of iterations per epoch."""
         iter_scheme = self.main_loop.data_stream.iteration_scheme
-        if hasattr(iter_scheme, 'num_batches'):
-            return iter_scheme.num_batches
+        if (hasattr(iter_scheme, 'indices') and
```

Comment:
What would be nice to have here for the sake of readability is the names of the iterations schemes that you have in mind for each of the branches.